docs/index.html

```diff
@@ -0,0 +1,189 @@
+			background: #ffffffbf;
+			color: #942cc8;
+		}
+
+		.label-description {
+			display: none;
+		}
+
+		label:hover .label-description {
+			display: inline-block;
+			position: absolute;
+			z-index: 999;
+			top: -100%;
+			left: 0;
+			background-color: #fff;
+			color: #000000;
+			padding: 0.25rem;
+			width: max-content;
+			font-size: 0.75rem;
+			line-height: 0.75rem;
+		}
+
+		.info-text {
+			padding: 2rem;
+			max-width: 600px;
+			margin: 0 auto;
+		}
+
+		a {
+			color: #942cc8;
+			text-decoration: none;
+		}
+
+		a:hover {
+			text-decoration: underline;
+			text-decoration-color: #000;
+		}
+
+		code {
+			background: #cecece;
+			padding: 0.25rem 0.35rem;
+			display: inline-block;
+		}
+
+		canvas {
+			max-width: 100%;
+			margin: 1rem auto;
+			display: block;
+		}
+
+		.header--inner-wrap {
+			max-width: 800px;
+			margin: 0 auto;
+		}
+	</style>
+</head>
+
+<body>
+	<header>
+		<div class="header--inner-wrap">
+			<h1>Shape Detection API</h1>
+
+			<label><input type="radio" name="detection-type" value="face" checked="checked"> Face detection </label>
+			<label><input type="radio" name="detection-type" value="barcode"> Barcode detection <span
+					class="label-description">Doesn't work reliably</span></label>
+			<label><input type="radio" name="detection-type" value="text"> Text detection <span
+					class="label-description">Works best with high contrast...</span></label>
+			<div class="upload-label">Analyze image file:</div>
+			<label class="upload-button"><input name="detection-file" type="file" accept="image/*">Choose file</label>
+		</div>
+	</header>
+
+	<div id="imageDisplay">
+		<div class="info-text">
+			<p>
+				This is the shape detection API implementation demo
+			</p>
+			<p>
+				The following demo allows you to detect shapes, barcodes and faces from an uploaded image.
+			</p>
+			<p>
+				Refer to the
+				<a href="https://ckeditor.com/blog/How-to-detect-human-faces-and-other-shapes-in-JavaScript/"
+					target="_blank">
+					How to detect human faces (and other shapes) in JavaScript
+				</a>
+				on the CKEditor Blog to find out more about the presented function.
+			</p>
+			<p>
+				Please bear in mind, this mechanism only works in Chrome and it requires this flag:
+				<code>chrome://flags#enable-experimental-web-platform-features</code>
+				to be turned on.
+			</p>
+			<p>
+
+			</p>
+		</div>
+		<canvas style="position: relative;" />
+	</div>
+	</div>
+	<script type="module">
+		import detectShape from './detector.mjs';
+		const el = document.querySelector('#imageDisplay');
+		const button = document.querySelector('input[type="file"]');
+		const canvas = document.querySelector("canvas");
+		const canvasContext = canvas.getContext('2d');
+		const infoText = document.querySelector(".info-text")
+
+		button.addEventListener('click', (event) => { event.target.value = null }) // Clearing the image - needed if you want to use the same one for different shape detections.
```

Comment:
Isn't it automatically done by the `change` listener?

Reply:
Nope, If the file is the same it doesn't trigger `change`